server/app/mutations/host_nodes/update.rb

```diff
@@ -4,19 +4,81 @@ module HostNodes
   class Update < Mutations::Command
 
     include Common
+    include Logging
 
     required do
       model :host_node
       array :labels, nils: true
     end
 
+    optional do
+      string :availability, in: ['active', 'drain']
+    end
+
     def execute
+      # Update node labels
       self.host_node.labels = self.labels if self.labels
       self.host_node.save
 
       notify_grid(self.host_node.grid)
 
-      self.host_node
+      # Update availability if given and trigger appropriate actions
+      if self.availability && self.host_node.availability != self.availability
+
+        self.host_node.set(:availability => self.availability)
+
+        case self.availability
+        when 'active'
+          start_stateful_services(self.host_node)
```

Comment:
If we want to really track the `drain` triggered deploys, IMO it has to be separate API (as it was in the beginning of this PR) that returns the deploy IDs created.